### PR TITLE
[build-tools] add support for Colcon build tool

### PR DESCRIPTION
**Pull request by seanyen**
This change is to add Colcon build tool support.

### CODE REVIEW

File: src/build-tool/colcon.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as vscode from "vscode"
```
Comment (kejxu) [inline src/build-tool/colcon.ts]: nit: add semicolon

File: src/build-tool/colcon.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as vscode from "vscode"
+import * as child_process from "child_process";
+
+import * as extension from "../extension"
```
Comment (kejxu) [inline src/build-tool/colcon.ts]: nit: add semicolon

File: src/build-tool/colcon.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Provides colcon build and test tasks.
+ */
+export class ColconProvider implements vscode.TaskProvider {
+    public provideTasks(token?: vscode.CancellationToken): vscode.ProviderResult<vscode.Task[]> {
+        let buildCommand: string;
+        let testCommand: string;
+
+        buildCommand = 'colcon build';
```
Comment (kejxu) [inline src/build-tool/colcon.ts]: maybe these can be initialized upon declaration:
```typescript
const buildCommand: string = "colcon build";
const testCommand: string = "colcon test";
```

**Comment by seanyen** [comment]
I cleaned up the code style by `tslint`.

**Pull request merged**